builtin/providers/librato/resource_librato_alert.go

```diff
@@ -0,0 +1,446 @@
+		for i, conditionDataM := range vs.List() {
+			conditionData := conditionDataM.(map[string]interface{})
+			var condition librato.AlertCondition
+			if v, ok := conditionData["type"].(string); ok && v != "" {
+				condition.Type = librato.String(v)
+			}
+			if v, ok := conditionData["threshold"].(float64); ok && !math.IsNaN(v) {
+				condition.Threshold = librato.Float(v)
+			}
+			if v, ok := conditionData["metric_name"].(string); ok && v != "" {
+				condition.MetricName = librato.String(v)
+			}
+			if v, ok := conditionData["source"].(string); ok && v != "" {
+				condition.Source = librato.String(v)
+			}
+			if v, ok := conditionData["detect_reset"].(bool); ok {
+				condition.DetectReset = librato.Bool(v)
+			}
+			if v, ok := conditionData["duration"].(uint); ok {
+				condition.Duration = librato.Uint(v)
+			}
+			if v, ok := conditionData["summary_function"].(string); ok && v != "" {
+				condition.SummaryFunction = librato.String(v)
+			}
+			conditions[i] = condition
+		}
+		alert.Conditions = conditions
+	}
+	if v, ok := d.GetOk("attributes"); ok {
+		attributeData := v.([]interface{})
+		if len(attributeData) > 1 {
+			return fmt.Errorf("Only one set of attributes per alert is supported")
+		} else if len(attributeData) == 1 {
+			if attributeData[0] == nil {
+				return fmt.Errorf("No attributes found in attributes block")
+			}
+			attributeDataMap := attributeData[0].(map[string]interface{})
+			attributes := new(librato.AlertAttributes)
+			if v, ok := attributeDataMap["runbook_url"].(string); ok && v != "" {
+				attributes.RunbookURL = librato.String(v)
+			}
+			alert.Attributes = attributes
+		}
+	}
+
+	alertResult, _, err := client.Alerts.Create(alert)
+
+	if err != nil {
+		return fmt.Errorf("Error creating Librato alert %s: %s", *alert.Name, err)
+	}
+
+	resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, _, err := client.Alerts.Get(*alertResult.ID)
+		if err != nil {
+			if errResp, ok := err.(*librato.ErrorResponse); ok && errResp.Response.StatusCode == 404 {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+
+	return resourceLibratoAlertReadResult(d, alertResult)
+}
+
+func resourceLibratoAlertRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*librato.Client)
+	id, err := strconv.ParseUint(d.Id(), 10, 0)
+	if err != nil {
+		return err
+	}
+
+	alert, _, err := client.Alerts.Get(uint(id))
+	if err != nil {
+		if errResp, ok := err.(*librato.ErrorResponse); ok && errResp.Response.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Librato Alert %s: %s", d.Id(), err)
+	}
+
+	return resourceLibratoAlertReadResult(d, alert)
+}
+
+func resourceLibratoAlertReadResult(d *schema.ResourceData, alert *librato.Alert) error {
+	d.SetId(strconv.FormatUint(uint64(*alert.ID), 10))
+	if alert.ID != nil {
+		if err := d.Set("id", *alert.ID); err != nil {
```

Comment:
A simplier way to do this is just

```
d.Set("id", alert.ID)
```

d.Set will work out if it is safe to dereference the value or not. This means we don't need to wrap it in a if != nil